packages/lwc/types.d.ts

```diff
@@ -178,4 +180,45 @@ declare module 'lwc' {
      * @param config configuration object for the accessor
      */
     export function wire(getType: (config?: any) => any, config?: any): PropertyDecorator;
+
+    type WireConfigValue = Record<string, any>;
+    type WireContextValue = Record<string, any>;
+
+    interface WireAdapter {
+        update(config: WireConfigValue, context?: WireContextValue): void;
+        connect(): void;
+        disconnect(): void;
+    }
+
+    type WireDataCallback = (value: any) => void;
+    type WireAdapterSchemaValue = 'optional' | 'required';
+    type ContextValue = Record<string, any>;
```

Comment:
would this one be the same as the `WireContextValue`? now that i see both (Wire..., Context...), it does makes more sense for it to be named `ContextValue`, what do you think about just using the later?

Reply:
@jodarove agree we can use `ContextValue` for both since they are the same, and so closely named